### PR TITLE
MultiplayerSpawner Refactor

### PR DESCRIPTION
**Pull request by thearst3rd**
Turns out this node is good

### CODE REVIEW

**Comment by thearst3rd** [comment]
Right now, this only handles the Player and not the Targets/Arrows yet. Those are to come, but I think it's in a state where it can be merged now so that our branches don't diverge too much.

File: src/objects/player.gd [hunk 1]
```diff
@@ -55,14 +50,34 @@ var previous_global_position: Vector3
...
-	# We want finer control of the camera node, so it gets set as a top level node with interpolation disabled
-	camera.set_as_top_level(true)
-	normal_fov = camera.fov
-	# camera.set_physics_interpolation_mode(Node.PHYSICS_INTERPOLATION_MODE_OFF)
+	var player_id := name.to_int()
+	if player_id == multiplayer.get_unique_id():
+		# We want finer control of the camera node, so it gets set as a top level node with interpolation disabled
+		camera.set_as_top_level(true)
+		normal_fov = camera.fov
+		# camera.set_physics_interpolation_mode(Node.PHYSICS_INTERPOLATION_MODE_OFF)
```
Comment (admoore0) [inline src/objects/player.gd:68]: Did you mean to leave this comment in?

File: src/objects/player.gd [hunk 2]
```diff
@@ -55,14 +50,34 @@ var previous_global_position: Vector3
...
+		$Nameplate.hide()
+		$BodyMesh.hide()
+		$Head/HeadMesh.hide()
+		$Camera3D.make_current()
```
Comment (admoore0) [inline src/objects/player.gd]: Just use `camera` here.

File: src/objects/player.gd [hunk 3]
```diff
@@ -273,14 +269,14 @@ func on_shot(body:Node) -> void:
 		var my_team := Multiplayer.get_player_by_id(name.to_int()).team_id
 		if shooter_team == my_team or not is_vulnerable:
 			body.queue_free()
-			get_parent().get_parent().arrow_collected(name)
+			owner.arrow_collected(name)
```
Comment (admoore0) [inline src/objects/player.gd:273]: What is `owner` here? Is it just a shorthand for the top level node?

File: src/states/menus/lobby.gd [hunk 4]
```diff
@@ -236,6 +236,7 @@ func generate_button_grid() -> void:
 		var button := preload("res://src/states/menus/buttons/color_button.tscn").instantiate() as ColorButton
 		button.set_button_color(COLORS[angle_idx])
 		button_circle.add_child(button)
+		button.owner = self
```
Comment (admoore0) [inline src/states/menus/lobby.gd:239]: What are these here in the lobby/scoreboard for. It doesn't look like they have an associated multiplayerspawner?

**Review by admoore0 — CHANGES_REQUESTED** [review]

File: src/objects/player.gd (hunk 1 above)
Comment (thearst3rd) [reply src/objects/player.gd:68]: Yes, Godot 3D physics interpolation isn't supported yet in Godot 4 but this line will be uncommented once it is.

**Review by admoore0 — APPROVED** [review]

**Review by kdreese — APPROVED** [review]

**Pull request merged**